src/style.rs

```diff
@@ -0,0 +1,234 @@
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+#[cfg(feature = "serde")]
+pub(crate) mod serde_support;
+
+/// The style of a [`Chat`]
+#[derive(Clone, Debug, Default, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Deserialize))]
+pub struct Style {
+    pub bold: Option<bool>,
+    pub italic: Option<bool>,
+    pub underlined: Option<bool>,
+    pub strikethrough: Option<bool>,
+    pub obfuscated: Option<bool>,
+    pub color: Option<ChatColor>,
+    /// This field is ignored for versions older than 1.8
+    pub insertion: Option<FrozenStr>,
+    /// This field is ignored for versions older than 1.16
+    pub font: Option<FrozenStr>,
+    #[cfg_attr(feature = "serde", serde(rename = "clickEvent"))]
+    pub click_event: Option<ClickEvent>,
+    #[cfg_attr(feature = "serde", serde(rename = "hoverEvent"))]
+    pub hover_event: Option<HoverEvent>,
+}
+
+impl Style {
+    pub fn new() -> Self {
+        Style::default()
+    }
+
+    pub fn and_color(mut self, color: Option<ChatColor>) -> Self {
+        self.color = color;
+        self
+    }
+
+    pub fn color(mut self, color: ChatColor) -> Self {
+        self.color = Some(color);
+        self
+    }
+
+    pub fn bold(mut self, bold: bool) -> Self {
+        self.bold = Some(bold);
+        self
+    }
+
+    pub fn italic(mut self, italic: bool) -> Self {
+        self.italic = Some(italic);
+        self
+    }
+
+    pub fn underlined(mut self, underlined: bool) -> Self {
+        self.underlined = Some(underlined);
+        self
+    }
+
+    pub fn strikethrough(mut self, strikethrough: bool) -> Self {
+        self.strikethrough = Some(strikethrough);
+        self
+    }
+
+    pub fn obfuscated(mut self, obfuscated: bool) -> Self {
+        self.obfuscated = Some(obfuscated);
+        self
+    }
+
+    pub fn font<T: Into<FrozenStr>>(mut self, font: Option<T>) -> Self {
+        self.font = font.map(|font| font.into());
+        self
+    }
+
+    pub fn insertion<T: Into<FrozenStr>>(mut self, insertion: Option<T>) -> Self {
+        self.insertion = insertion.map(|insertion| insertion.into());
+        self
+    }
+
+    pub fn click(mut self, click_event: Option<ClickEvent>) -> Self {
+        self.click_event = click_event;
+        self
+    }
+
+    pub fn hover(mut self, hover_event: Option<HoverEvent>) -> Self {
+        self.hover_event = hover_event;
+        self
+    }
+}
+
+/// The different colors a [`Chat`] can have.
+/// ## TODO: Automatically find nearest value when serializing [`ChatColor::Custom`] for older versions
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum ChatColor {
+    Black,
+    DarkBlue,
+    DarkGreen,
+    DarkCyan,
+    DarkRed,
+    Purple,
+    Gold,
+    Gray,
+    DarkGray,
+    Blue,
+    Green,
+    Cyan,
+    Red,
+    Pink,
+    Yellow,
+    White,
+    /// This field is ignored for versions older than 1.16.
+    ///
+    /// See [`ChatColor::custom()`].
+    Custom(FrozenStr),
```

Comment:
Won't be better if we base `ChatColor::Custom` on `(u8, u8, u8)`? It will allow for example `Copy` trait on it

Reply:
That's all in the todo at the top of the color serialization, I recommend opening a different pr for the colors.